embedded/store/immustore.go

```diff
@@ -2036,7 +2259,18 @@ func (s *ImmuStore) appendableReaderForTx(txID uint64) (*appendable.Reader, erro
 }
 
 func (s *ImmuStore) ReadTx(txID uint64, tx *Tx) error {
-	r, err := s.appendableReaderForTx(txID)
+	s.mutex.Lock()
```

Comment:
probably it can be removed, as appendable and cache are thread-safe, and the state-related attributes have its own lock... I don't expect an noticeable impact in performance, it will ensure if any read is initiated it will complete and not being closed after partially reading...